Odata-docs/webapi-8/fundamentals/query-options.md

```diff
@@ -0,0 +1,486 @@
+---
+title: "OData query Options"
+description: "This documentation describes how to use the supported query options."
+
+author: KenitoInc
+ms.author: kemunga
+
+ms.date: 12/16/2022
+---
+# OData query options
+**Applies To**:[!INCLUDE[appliesto-webapi](../../includes/appliesto-webapi-v8.md)]
+
+AspNetCore OData 8.x provides out of the box querying capabilities.
+
+A query option is basically requesting that a service perform a set of transformations such as filtering, sorting, etc. to its data before returning the results. See more information [here](/odata/concepts/queryoptions-overview).
+
+## Model class
+We will define our model as follows:
+
+```csharp
+public class Customer
+{
+    [Key]
+    public int Id { get; set; }
+    public String Name { get; set; }
+    public int Age { get; set; }
+    public List<Order> Orders { get; set; }
+}
+
+public class Order
+{
+    [Key]
+    public int Id { get; set; }
+    public int Price { get; set; }
+    public int Quantity { get; set; }
+}
+```
+
+We have a `Customer` entity with `Orders` navigation property.
+
+## Configuring Startup class
+```csharp
+public class Startup
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        var modelBuilder = new ODataConventionModelBuilder();
+        modelBuilder.EntitySet<Customer>("Customers");
+
+        services.AddControllers().AddOData(
+            options => options.Select().Filter().OrderBy().Expand().Count().SetMaxTop(null).AddRouteComponents(
```

Comment:
One OData query feature that get forgotten a lot is skip token - `SkipToken()`. I think you should talk a little about it. It might even deserve a doc of it's own

Reply:
I deliberately left it since we will have a separate pagination doc